app/src/main/java/app/musikus/ui/activesession/ActiveSessionUiState.kt

```diff
@@ -8,20 +8,144 @@
 
 package app.musikus.ui.activesession
 
-import app.musikus.database.Nullable
-import app.musikus.database.daos.LibraryFolder
+import androidx.compose.runtime.Stable
+import androidx.compose.ui.graphics.Color
+import app.musikus.database.LibraryFolderWithItems
+import app.musikus.database.UUIDConverter
 import app.musikus.database.daos.LibraryItem
-import app.musikus.ui.library.DialogMode
-import app.musikus.ui.library.LibraryItemDialogUiEvent
-import app.musikus.ui.library.LibraryItemDialogUiState
-import app.musikus.ui.library.LibraryItemEditData
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
 import java.time.ZonedDateTime
 import java.util.UUID
-import kotlin.time.Duration
 
 
+// TODO UIText for all Strings?
 
+@Stable
+enum class SessionPausedResumedState {
+    RUNNING,
+    PAUSED
+}
+
+/** Specific one-time events triggering some UI action in ActiveSession */
+@Stable
+data class ActiveSessionEventStates (
+    val sessionSaved: Boolean = false,
+    val sessionDiscarded: Boolean = false,
+)
+
+@Stable
+/** Main UI State */
 data class ActiveSessionUiState(
+    val topBarUiState: StateFlow<ActiveSessionTopBarUiState> = MutableStateFlow(ActiveSessionTopBarUiState()),
+    val mainContentUiState: StateFlow<MainContentUiState> = MutableStateFlow(MainContentUiState()),
+    val newItemSelectorUiState: StateFlow<NewItemSelectorUiState> = MutableStateFlow(NewItemSelectorUiState()),
+    val toolsUiState: StateFlow<ActiveSessionToolsUiState> = MutableStateFlow(ActiveSessionToolsUiState()),
+    val dialogVisibilities: StateFlow<DialogVisibilities> = MutableStateFlow(DialogVisibilities()),
+)
+
+data class DialogVisibilities(
+    val newItemSelectorVisible: Boolean = false,
+    val finishDialogVisible: Boolean = false,
+    val discardDialogVisible: Boolean = false,
+    val createItemDialogVisible: Boolean = false,
+    val createFolderDialogVisible: Boolean = false,
+)
+
+@Stable
+data class MainContentUiState(
+    val timerUiState: StateFlow<ActiveSessionTimerUiState> = MutableStateFlow(ActiveSessionTimerUiState()),
+    val currentItemUiState: StateFlow<ActiveSessionCurrentItemUiState> = MutableStateFlow(ActiveSessionCurrentItemUiState()),
+    val pastSectionsUiState: StateFlow<ActiveSessionCompletedSectionsUiState> = MutableStateFlow(ActiveSessionCompletedSectionsUiState()),
+    val endDialogUiState: StateFlow<ActiveSessionEndDialogUiState> = MutableStateFlow(ActiveSessionEndDialogUiState())
```

Comment:
Having a `visible` flag has the advantage that the comments and rating a user maybe wrote in the field stays saved after the user dismisses the dialog. This way could maybe add some comments along the way and they all stay in the dialog when it is opened.

Reply:
Touché if that is indeed the desired behavior.  🤔

Reply:
Actually, no. Just because the UiState is null doesn't mean we can't store the comment in the ViewModel

Reply:
Yes, maybe it would be better anyways to make it properly and store it somewhere more persistent than in the ViewModel (which will be destroyed on app close)

Reply:
In any case, the `visible` flag is kind of redundant and in the case it is `false`, all other properties values are irrelevant so the whole object might as well be `null`